### PR TITLE
Guus/rust tracy 2

### PR DESCRIPTION
**Pull request by guusw**

### CODE REVIEW

File: cmake/Platform.cmake [hunk 1]
```diff
@@ -2,6 +2,10 @@ if(UNIX AND NOT(APPLE OR ANDROID OR EMSCRIPTEN))
   set(DESKTOP_LINUX TRUE)
 endif()
 
+# set(CMAKE_OBJECT_PATH_MAX 512)
```
Comment (sinkingsugar) [inline cmake/Platform.cmake]: Remove?

File: cmake/rust/union_lib.rs.in [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// This file is generated by CMake
```
Comment (sinkingsugar) [inline cmake/rust/union_lib.rs.in:1]: Says generated? But we commit it?

Overall this should ideally not be generated, the header part I mean should be editable and cmake concats

File: shards/modules/spatial/CMakeLists.txt [hunk 3]
```diff
@@ -8,5 +8,5 @@
 #   REGISTER_SHARDS spatial
 #   EXPERIMENTAL)
 
-# target_link_libraries(shards-module-spatial gfx input shards-module-egui)
+# target_link_libraries(shards-module-spatial gfx-core input shards-module-egui)
```
Comment (sinkingsugar) [inline shards/modules/spatial/CMakeLists.txt:11]: Remove? Or just disabled for now I guess?

**Review by sinkingsugar — APPROVED** [review]
Just some minor stuff.

But before merge you should start a document somewhere explaining the rationale a bit, just in case we scale and someone else needs to change things. Will avoid confusion.
Also check Xcode iOS build.

File: cmake/Modules.cmake [hunk 4]
```diff
@@ -32,13 +34,17 @@ function(add_shards_module MODULE_NAME)
...
     if(NOT EXISTS ${DUMMY_SOURCE_PATH})
       file(WRITE ${DUMMY_SOURCE_PATH} "// Intentionally empty\n")
     endif()
+
     set(MODULE_SOURCES ${DUMMY_SOURCE_PATH})
   endif()
 
-  add_library(${MODULE_TARGET} STATIC ${MODULE_SOURCES})
+  add_library(${MODULE_TARGET} OBJECT ${MODULE_SOURCES})
+  # add_library(${MODULE_TARGET}-link INTERFACE)
```
Comment (guusw) [inline cmake/Modules.cmake]: Cleanup

File: cmake/rust/union_lib.rs.in (hunk 2 above)
Comment (guusw) [reply cmake/rust/union_lib.rs.in:1]: Yeah it's a template ;)

File: shards/modules/spatial/CMakeLists.txt (hunk 3 above)
Comment (guusw) [reply shards/modules/spatial/CMakeLists.txt:11]: Was already disabled, just fixed the reference with find-replace

**Pull request merged**